### DISCUSSION TITLE
How to run ITF with code coverage?

### DISCUSSION THREAD

**Discussion by rakus** [khmarbaise/maven-it-extension#328] [Q&A]
Hello,

I was trying to run itf integration tests with jacoco code coverage. 

First I configured the failsafe plugin with the argLine created by the jacoco-maven-plugin:preparte-agent. Similar to code coverage for unit tests. That didn't work.

After a lot of trying around and hacking I finally got it running by _manually_ creating the jacoco javaagent command line and configure it as environment variable `MAVEN_OPTS` in the failsafe plugin.

See this [commit](https://github.com/rakus/properties-constants-maven-plugin/commit/0edfab0a2e52bc4788f3f714873865aa4e7c2830).

It works, but I classify that as a "hack" and I'm uncomfortable with it.

Any ideas? Did I miss anything?

Ralf

**Comment by khmarbaise** [comment]
That approach looks interesting... I will take a deeper look into it... Thanks for offering this here...

**Comment by bitcoder** [comment]
works for me. thanks for sharing @rakus !

**Comment by marwin1991** [comment]
thanks, for anyone looking for multimodule maven project example:

https://github.com/logchange/logchange/pull/663/commits/41b8c53f58b75b0920762f169a358f9fc90b3ebe
